### ISSUE TITLE
Hide magic lines in slides?

### ISSUE THREAD

**Issue by abingham** [damianavila/RISE#236]
Would it be possible (or is there already a way) to hide magics in slides? For example, I'm using `ipytest` to demo pytest in a notebook slideshow, and all of my demo slides look like:
```
%%run_pytest -q --disable-pytest-warnings

def test_something():
    . . . blah blah blah . . .
```

The magic has to be in each cell to function, but it also clutters up the slides and feels unnecessary. So I'd like it if there were some way to hide the magic slides.

Perhaps there's some CSS trick I could use, and that would be perfectly fine. But if not, and if it wouldn't be too hard to add to RISE, I think it would be a great addition.

**Comment by damianavila** [comment]
Since the magic is in the cell content, I don't see an easy way to hide it.
We could eventually think in some kind of workaround but not sure how valuable would be this "hide" option for others, so a native solution would have a low priority, I guess...

**Comment by parmentelat** [comment]
I would agree that hiding all magics in all cells would be going a little too far, as there are many cases where removing them could create confusion for readers

**Comment by abingham** [comment]
> there are many cases where removing them could create confusion for readers

I completely agree. If there's any support for hiding magic, it should be optional and off by default.

**Comment by ziyuang** [comment]
Is there any progress on hiding cells from a presentation but still running them?

**Comment by parmentelat** [comment]
within nbclassic there are - or at least there used to be - a hide-input extension that allowed to declare a cell as 'hide-input' through its metadata
I remember having used this already within rise slideshows, and I expect it should still work; I don't remember the actual details though...

**Comment by parmentelat** [comment]
no I was meaning this one:
https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/hide_input/readme.html
